app/controllers/datasets_controller.rb

```diff
@@ -1,2 +1,107 @@
 class DatasetsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @dataset = Dataset.new
+  end
+
+  def create
+    @dataset = Dataset.new(params.require(:dataset).permit(:id, :title, :summary, :description))
+    @dataset.organisation = current_user.primary_organisation
+
+    if @dataset.save
```

Comment:
We should probably slugify the title into the dataset.name field here as the end url will be /dataset/NAME